src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -779,7 +779,7 @@ protected ArgumentsAndBlockTranslation translateArgumentsAndBlock(SourceIndexLen
             throw new UnsupportedOperationException("Unknown argument node type: " + argsNode.getClass());
         }
 
-        final RubyNode[] argumentsTranslated = new RubyNode[arguments.length];
+        final RubyNode[] argumentsTranslated = arguments.length == 0 ? RubyNode.EMPTY_ARRAY : new RubyNode[arguments.length];
```

Comment:
Could you add a helper method like `createRubyNodeArray(int size)` to share the logic?

Reply:
fixed, create `RubyNode.createArray(int size)`